src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/sql.js

```diff
@@ -0,0 +1,274 @@
+        }
+        .tooltip {
+            font-family: DejaVuSansMonoBook, sans-serif;
+            font-size: 12px;
+            opacity: 0;
+            transition: opacity 0.2s;
+            display: inline-block;
+            white-space: nowrap;
+            background-color: rgba(249, 249, 249, 1);
+            box-shadow: 0 0 16px rgba(0, 0, 0, 0.16);
+            text-align: left;
+            border-radius: 6px;
+            padding: 5px;
+            position: absolute;
+            z-index: 99999;
+            pointer-events: none;
+        }
+        </style>
+        `
+
+        function renderColor(color) {
+            return 'rgba(' + color[0] + ',' + color[1] + ',' + color[2] + ',' + color[3] + ')'
+        }
+
+        const bgAlpha = 0.25
+        const theme = this.theme
+
+        function splitTypeName(name) {
+            var ix = name.lastIndexOf('.')
+            if (ix < 0) {
+                return ['', name]
+            }
+
+            return [name.substr(0, ix + 1), name.substr(ix + 1)]
+        }
+
+        function renderParameter(param) {
+            const actualType = param.actual_type
+            let value = param.value
+
+            if (actualType == 'Builtins.Main.Text') {
+                value = "'" + value.replaceAll("'", "''") + "'"
+            }
+
+            const actualTypeColor = theme.getColorForType(actualType)
+            const fgColor = actualTypeColor
+            let bgColor = [...fgColor.slice(0, 3), bgAlpha]
+            const expectedEnsoType = param.expected_enso_type
+
+            let html = ''
+            if (actualType == expectedEnsoType) {
+                const elem = document.createElement('div')
+                html +=
+                    '<div class="interpolation" style="color:' +
+                    renderColor(fgColor) +
+                    ';background-color:' +
+                    renderColor(bgColor) +
+                    ';">'
+                html += value
+                html += '</div>'
+            } else {
+                let expectedType = expectedEnsoType
+                if (expectedType === null) {
+                    expectedType = 'Standard.Database.Data.Sql.Sql_Type.' + param.expected_sql_type
+                }
+
+                const expectedTypeColor = theme.getColorForType(expectedType)
+                const hoverBgColor = expectedTypeColor
+                bgColor = [...hoverBgColor.slice(0, 3), bgAlpha]
+                const hoverFgColor = theme.getForegroundColorForType(expectedType)
+
+                const received = splitTypeName(actualType)
+                const expected = splitTypeName(expectedType)
+                let message =
+                    'Received <span class="modulepath">' +
+                    received[0] +
+                    '</span><span style="color: ' +
+                    renderColor(actualTypeColor) +
+                    '">' +
+                    received[1] +
+                    '</span><br>'
```

Comment:
these are VERY complex expressions. All such multi-line expressions (in the whole new code) please split into well named variables - semantically named. So do `let someMeaningfulNAme = received[0]` etc.